samples/nrf9160/gps/src/main.c

```diff
@@ -410,40 +337,61 @@ int main(void)
 	}
 #endif
 
-	printk("Getting GPS data...\n");
+	printk("Getting GNSS data...\n");
 
 	while (1) {
-
-		do {
-			/* Loop until we don't have more
-			 * data to read
-			 */
-		} while (process_gps_data(&gps_data) > 0);
-
 		if (IS_ENABLED(CONFIG_GPS_SAMPLE_NMEA_ONLY)) {
 			print_nmea_data();
-			nmea_string_cnt = 0;
 		} else {
+			if (new_data) {
```

Comment:
Could we use a semaphore or something instead for this synchronization so that we don't need to loop, but rather wait for new data? `k_poll` API can be used to wait for both sempahore and new data in message queue at the same time: https://docs.zephyrproject.org/latest/reference/kernel/other/polling.html

Reply:
Added a new semaphore and use `k_poll()` to wait for either new PVT or NMEA data.